styx-scheduler-service/src/test/java/com/spotify/styx/SchedulerTest.java

```diff
@@ -87,8 +87,19 @@ public void shouldNotTransitionIfNotTimedOut() throws Exception {
   }
 
   @Test
-  public void shouldTriggerRetryIfDelayHasPassed() throws Exception {
-    StateData stateData = StateData.builder().retryDelayMillis(15_000).build();
+  public void shouldExecuteRetryIfDelayHasPassed() throws Exception {
+    StateData stateData = StateData.builder().retryDelayMillis(15_000).tries(10).build();
+    setUp(20, RunState.create(INSTANCE, State.QUEUED, stateData, time));
+
+    now = now.plus(15, ChronoUnit.SECONDS);
+    scheduler.tick();
+
+    assertThat(stateManager.get(INSTANCE).state(), is(State.PREPARE));
+  }
+
+  @Test
+  public void shouldExecuteNewTriggers() throws Exception {
+    StateData stateData = StateData.builder().tries(0).build();
     setUp(20, RunState.create(INSTANCE, State.QUEUED, stateData, time));
 
     now = now.plus(15, ChronoUnit.SECONDS);
```

Comment:
Maybe this line should be removed so that we ensure that it dequeues the state immediately?